Bug 1779585 - Cast version numbers to a number.

denschub opened the pull request:
Otherwise, this would compare strings to each other, and `MAX('100', '99')` is actually `99`.

Note that I can't really test this. I ran the query in BigQuery, and didn't see versions larger than 99 with the old query, but I do see them with the cast. I'm not sure if this actually fixes the public data report, but maybe it does. If someone could help verify that, that'd be awesome.

denschub commented:
For reference, that's the result of the `latest_version_per_day` subquery, with the old one on the left, and the new one on the right.

That's the best testing I can do with my current knowledge. :)

<img width="1493" alt="Screenshot 2023-01-14 at 04 30 42" src="https://user-images.githubusercontent.com/344777/212448830-a842d6e7-9ad2-4df0-811c-289f73ac91c5.png">

akkomar reviewed (changes requested):
Thanks for taking this on @denschub !

`public_data_report_user_activity` query is failing for me with `No matching signature for operator = for argument types: STRING, NUMERIC. Supported signature: ANY = ANY at [190:29]` error. I think we need to normalize `major_version` in `active_clients_weekly` the same way.

We should also use ``` `mozfun.norm.truncate_version\`(app_version, "major")``` for normalization here (see https://docs.telemetry.mozilla.org/cookbooks/looker/browser_versions.html#normalizing-browser-version-data).

akkomar commented:
@denschub  I can put up another PR with fixes if you prefer.

denschub commented:
Ah, right, thanks for reminding me that that function exists, @akkomar. :) I've updated the PR accordingly, and also normalized `active_clients_weekly` the same way.

Does that work better for you? (It seems like if I try to run that SQL, all I get is a "Cannot query over table 'moz-fx-data-shared-prod.telemetry_derived.clients_last_seen_v1' without a filter over column(s) 'submission_date' that can be used for partition elimination" - which seems unrelated to what I changed.)
